js/kanji.js

```diff
@@ -150,6 +151,52 @@ window.Kanji =  {
     }
   },
 
+  _initialSelectAllHide: function() {
+    $(".category-select-all").each(function() {
+      $(this).hide();
+    });
+    $(".category-deselect-all").each(function() {
+      $(this).hide();
+    });
+    $(".category-select-all-show").each(function() {
+      $(this).show();
+    });
+  },
+
+  __selectClicker: function() {
+    if (!$(this).hasClass('selected')) {
+      $(this).trigger("click");
+    }
+  },
+
+  __deselectClicker: function() {
+    if ($(this).hasClass('selected')) {
+      $(this).trigger("click");
+    }
+  },
+
+  _handleSelectAll: function() {
+    $(".category-select-all,.category-deselect-all").off("click"); // remove previous event handlers
+
+    $(".category-select-all").on('click', function() {
+      $(this).siblings(".category-content").children(".kanji-box").each(Kanji.__selectClicker);
+      $(this).siblings(".kanji-box").each(Kanji.__selectClicker);
+      $(this).removeClass("category-select-all");
+      $(this).addClass("category-deselect-all");
+      $(this).children("a").html("De-select All");
+      Kanji._handleSelectAll(); // need to readd event handlers
+    });
+
+    $(".category-deselect-all").on('click', function() {
+      $(this).siblings(".category-content").children(".kanji-box").each(Kanji.__deselectClicker);
+      $(this).siblings(".kanji-box").each(Kanji.__deselectClicker);
+      $(this).removeClass("category-deselect-all");
+      $(this).addClass("category-select-all");
+      $(this).children("a").html("Select All");
```

Comment:
I think it would be easier to modify the user interface later if we try to keep all of the content in the DOM and out of the Javascript. 

So if we wanted to change this copy we would have to change the javasript instead of the DOM. Maybe this would hide/show the link instead of changing it's contents.